frontend/src/pages/home-page.tsx

```diff
@@ -52,9 +53,11 @@ function HomePage() {
         <BlogFeed />
         <h1 className="text-2xl font-semibold">All Blog Posts</h1>
         <div className="-mx-4 flex flex-wrap">
-          {posts.map((post) => (
-            <PostCard key={post._id} post={post} />
-          ))}
+          {posts.length === 0
+            ? Array(8)
+                .fill(0)
+                .map((_, index) => <PostCardSkeleton key={index} />)
+            : posts.map((post) => <PostCard key={post._id} post={post} />)}
```

Comment:
Same here , one for `Array(8)
                .fill(0)
                .map((_, index) => <PostCardSkeleton key={index} />)` So that we can make it more readable 
                
We can use this following logic, (My approach, feel free to enhance and add your touch if better)
Creating a new function which will take 3 params
1.  The collections like `latestPosts`, `posts`  
2.  The type of collection , we can have a enum for this `HOME_POST`, `FEATURED_POST` & `LATEST_POST` 
3.  `isSkeleton` of type `bool` if yes we can render the corresponding `skeleton` or the actual content using the switch case

We can do this in new issue, so that we can seperate the refactoring from this feature implementation

Reply:
Hii @krishnaacharyaa, I think to do this switch case logic we do need to maintain state variable (lets say isLoading) before api call and after api call right??
Okay, we can do this in new issue...

Reply:
@hemant14050  Ideally yes, we can use the context api for introducing the same, for much cleaner approach. But for now we can create refactoring in new issue with `posts.length==0` and using the context api in another issue